src/main/scala/io/iohk/ethereum/domain/Account.scala

```diff
@@ -42,6 +42,9 @@ case class Account(
   def withStorage(storageRoot: ByteString): Account =
     copy(storageRoot = storageRoot)
 
+  def clearAccount: Account =
+    copy(nonce = UInt256.Zero, storageRoot = Account.EmptyStorageRootHash, codeHash = Account.EmptyCodeHash)
```

Comment:
How is it different from `Account.empty()`?

Reply:
`Account.empty()` creates account with `balance = 0`, `clearAccount` has somewhat unintuitive behaviour that it leaves account ether balance intact, but clears other fields.

Reply:
The name of this method is misleading (think about a bank). Also the nonce should not be just set to zero but to the configured value (as with `Account.empty`).

Also are we sure we need to reset the nonce? In the event of a collision this is very likely to lead to more collisions.

Reply:
Actually, after EIP161 it should be changed to `1`, but now `0` is correct.  (https://github.com/ethereum/cpp-ethereum/issues/4225)